sdk/Sdk.Generators/FunctionMetadataProviderGenerator/ReferencedAssemblyMethodVisitor.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Generators
+{
+    /// <summary>
+    /// Visits all symbols from referenced assemblies and returns all methods which are valid Azure Functions.
+    /// </summary>
+    internal sealed class ReferencedAssemblyMethodVisitor : SymbolVisitor
+    {
+        private readonly Compilation _compilation;
+
+        /// <summary>
+        /// Gets all methods which are valid Azure Functions.
+        /// </summary>
+        internal readonly List<IMethodSymbol> FunctionMethods = new();
+
+        internal ReferencedAssemblyMethodVisitor(Compilation compilation)
+        {
+            _compilation = compilation ?? throw new ArgumentNullException(nameof(compilation));
+        }
+
+        public override void VisitModule(IModuleSymbol moduleSymbol)
+        {
+            foreach (var assemblySymbol in moduleSymbol.ReferencedAssemblySymbols)
```

Comment:
This code can be updated to get the methods from the entry assembly as well. But did not want to do that in this PR to keep the scope of the changes from this PR to minimum